lib/mini_magick/image.rb

```diff
@@ -613,5 +635,42 @@ def landscape?
     def portrait?
       height > width
     end
+
+    ##
+    # As a security precaution, ensure the attacker cannot execute arbitrary code by
+    # (for example) uploading a carefully crafted Postscript with a JPEG file extension.
+    # Forces the input file format by specifying a module prefix on the filename, such as
+    # "jpg:some-file.jpg"
+    #
+    def self.force_input_format?(path_or_url)
+      prefix_colon = (%r{\A[A-Za-z][A-Za-z0-9+\-\.]*:} =~ path_or_url)
+      uri_colon = (%r{\A[A-Za-z][A-Za-z0-9+\-\.]*://} =~ path_or_url)
+      if (prefix_colon && !uri_colon) #prefix is present before colon and is not a URI
+        true
+      else
+        false
+      end
+    end
```

Comment:
@janko Thank you for the feedback. I tried to differentiate between the typical URI colon and the security prefix colon in this function. Please let me know if you would prefer a different approach or if you would like to see some specific test coverage.

Reply:
How I imagined this ImageMagick feature to be used is that we would read the file extension, and then use that as the type when processing. So, if I open an Image with `.jpg` extension, then when I'm processing, instead of `mogrify filename.jpg -resize 300x300` being executed, we'd executed `mogrify jpg:filename.jpg -resize 300x300`.

In other words, instead of checking for mismatches on our end, we choose a more declarative approach of interpreting the image in the expected format. That would ideally happen after an URI has been resolved and a tempfile is given, so that no matter if you're using `MiniMagick::Image.open("/path/to/file")` or `MiniMagick::Image.open("https://example.com/path/to/file")`, it works without having to handle any URIs.

Reply:
So, I didn't imagine any checks to be performed on `MiniMagick::Image` initialization, but on future processing.

Reply:
Thanks for the feedback. I believe I understand the requirements. I'll try again from this angle 👍